components/board/Article.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useCallback, useState } from 'react';
+import { useRouter } from 'next/router';
+import * as BS from '@/components/board/Styled';
+import LikeIcon from '@/components/common/images/ic_heart.png';
+import ProfileImg from '@/components/common/images/profile.png';
+import { getArticleDetail } from '../api/api';
+import formatDate from '../common/function/formatDate';
+export interface Article {
+    id: number;
+    title: string;
+    content: string;
+    images: string;
+    writer: Writer;
+    likeCount: number;
+    createdAt: string;
+}
+
+export interface Writer {
+    nickname: string;
+    id: number;
+}
+
+export default function Article() {
+    const router = useRouter();
+    const id = router.query['id'];
+    const [item, setItem] = useState<Article | undefined>(undefined);
```

Comment:
P3:
`undefined`는 명시해주지 않으셔도 됩니다.

```suggestion
    const [item, setItem] = useState<Article>();
```